### PR TITLE
runtime: Account nicknames

### PR DESCRIPTION
**Pull request by gavofyork**
Simple "nicknames" module for attaching names to accounts.

### CODE REVIEW

File: srml/nicks/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,331 @@
+// Copyright 2017-2019 Parity Technologies (UK) Ltd.
```
Comment (bkchr) [inline srml/nicks/src/lib.rs]:
```suggestion
// Copyright 2019 Parity Technologies (UK) Ltd.
```

File: srml/nicks/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+//! ## Overview
+//!
+//! Nicks is a trivial module for keeping track of account names on-chain. It makes no effort to
+//! create a name hierarchy, be a DNS replacement or provide reverse lookups.
+//!
+//! ## Interface
+//!
+//! ### Dispatchable Functions
+//!
+//! Only the sudo key can call the dispatchable functions from the Sudo module.
```
Comment (bkchr) [inline srml/nicks/src/lib.rs]: Still mentions `sudo`.

File: srml/nicks/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+		///
+		/// The dispatch origin for this call must be _Root_ or match `T::KillOrigin`.
+		///
+		/// # <weight>
+		/// - O(1).
+		/// - Limited storage reads.
+		/// - One DB write (event).
+		/// - Unknown weight of derivative `proposal` execution.
+		/// # </weight>
+		#[weight = SimpleDispatchInfo::FixedOperational(0)]
```
Comment (kianenigma) [inline srml/nicks/src/lib.rs]:
```suggestion
		#[weight = SimpleDispatchInfo::FreeOperational]
```

File: srml/nicks/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,331 @@
...
+		///
+		/// If the account doesn't already have a name, then a fee of `ReservationFee` is reserved
+		/// in the account.
+		///
+		/// The dispatch origin for this call must be _Signed_.
+		///
+		/// # <weight>
+		/// - O(1).
+		/// - Limited storage reads.
+		/// - One DB write (event).
```
Comment (bkchr) [inline srml/nicks/src/lib.rs]: We have a second write for inserting the name.

File: srml/nicks/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+		}
+
+		/// Clear an account's name and return the deposit. Fails if the account was not named.
+		///
+		/// The dispatch origin for this call must be _Signed_.
+		///
+		/// # <weight>
+		/// - O(1).
+		/// - Limited storage reads.
+		/// - One DB change.
```
Comment (bkchr) [inline srml/nicks/src/lib.rs]: We are writing an event and taking the name.

**Comment by gavofyork** [comment]
> Looks good, but don't we need to implement OnFreeBalanceZero to clear the nickname?

As long as ReservationFee is greater than ExistentialDesposit (will add the note to the docs), then no, since there will always need to be at least that reserved.

**Comment by bkchr** [comment]
Ahh okay, makes sense :)

**Pull request merged**